### DISCUSSION TITLE
Incorrect detection of sentence boundaries, if last sentence missing eos symbol for trf model

### DISCUSSION THREAD

**Discussion by koder-ua** [explosion/spaCy#13356] [Help: Other Questions] [answered]
## How to reproduce the behaviour
```
In [69]: len(list(spacy.load("en_core_web_trf")("The first sentence. The second sentence. The last one").sents))
Out[69]: 1   <<<<<<<<<<<<<<<<<<<<<< WRONG

In [70]: len(list(spacy.load("en_core_web_trf")("The first sentence. The second sentence. The last one.").sents))
Out[70]: 3

In [71]: len(list(spacy.load("en_core_web_sm")("The first sentence. The second sentence. The last one").sents))
Out[71]: 3

In [72]: len(list(spacy.load("en_core_web_sm")("The first sentence. The second sentence. The last one.").sents))
Out[72]: 3
```

## Your Environment
* Operating System: max os x 10.3
* Python Version Used: 3.11
* spaCy Version Used: 3.7.4
* Environment Information:
```
en_core_web_trf.__version__  >> '3.7.3'
en_core_web_sm.__version__ >> '3.7.1'
```

**Comment by svlandeg** [comment, marked as answer]
Hi!

In this pretrained pipeline, the sentence segmentation is actually done by the [parser](https://spacy.io/usage/linguistic-features#sbd-parser), and the model was mostly trained on texts with correct punctuation. So unfortunately this type of occassional error is unavoidable. 

If you'd like to have more predictable behaviour, you can use the [`sentencizer`](https://spacy.io/usage/linguistic-features#sbd-component) instead, which is a more simple rule-based component that splits sentences on punctuation like `.`, `!` or `?`.
